ethdb/pebble/pebble.go

```diff
@@ -121,12 +122,14 @@ func (d *Database) onCompactionBegin(info pebble.CompactionInfo) {
 }
 
 func (d *Database) onCompactionEnd(info pebble.CompactionInfo) {
-	if d.activeComp == 1 {
+	switch d.activeComp {
+	case 1:
 		d.compTime.Add(int64(time.Since(d.compStartTime)))
-	} else if d.activeComp == 0 {
+	case 0:
 		panic("should not happen")
+	default:
+		d.activeComp--
```

Comment:
should bring this out of the `switch` statement.  Otherwise, the compaction counter logic is broken.